### PR TITLE
Implement signmessagewithprivkey JSON-RPC command (rpc server)

### PR DESCRIPTION
**Pull request by lindlof**
Reuse the Bitcoin message signature header const
also in verifymessage.

### CODE REVIEW

**Comment by lindlof** [comment]
Because of this Satoshi was finally able to [sign his message](https://www.coindesk.com/craig-wright-called-fraud-message-signed-bitcoin-addresses-satoshi) the next day 😂

File: rpcserver.go [hunk 1]
```diff
@@ -3435,6 +3436,46 @@ func handleSetGenerate(s *rpcServer, cmd interface{}, closeChan <-chan struct{})
...
 
+// Text used to signify that a signed message follows and to prevent
+// inadvertently signing a transaction.
+const messageSignatureHeader = "Bitcoin Signed Message:\n"
+
+// handleSignMessageWithPrivKey implements the signmessagewithprivkey command.
+func handleSignMessageWithPrivKey(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
+	c := cmd.(*btcjson.SignMessageWithPrivKeyCmd)
+
+	wif, err := btcutil.DecodeWIF(c.PrivKey)
```
Comment (onyb) [inline rpcserver.go:3447]: `bitcoind` rejects WIFs that do not belong to the current network. Or so it seems. Here's an example for you to reproduce using a mainnet WIF:

```sh
$ ./btcctl --regtest --rpcuser=aaaa --rpcpass=bbbb signmessagewithprivkey 5HueCGU8rMjxEXxiPuD5BDku4MkFqeZyd4dZ1jvhTVqvbTLvyTJ iamsatoshi
G+brEqKGQgPCSkPJrBau7x32eFhT3Y5YXdEXKG2bg2E2Tqi0lU54t+rYrEezLcSDLS4TM+/AO8a3A9ybAfvyMLY=

$ bitcoin-cli -regtest signmessagewithprivkey 5HueCGU8rMjxEXxiPuD5BDku4MkFqeZyd4dZ1jvhTVqvbTLvyTJ iamsatoshi
error code: -5
error message:
Invalid private key
```

The fix would be to simply check `wif.IsForNet(s.cfg.ChainParams)`.

Comment (lindlof) [reply rpcserver.go:3447]: Yes this was the case. Fixed by checking `wif.IsForNet(s.cfg.ChainParams)` as you suggested.

**Review by onyb — APPROVED** [review]
Tested OK locally. ✔️

**Comment by jakesyl** [comment]
Looks good to me/tested locally against 0.20.0/0.19.0 (on bitcoind side). Since I'm not as familiar with [btcec](https://github.com/btcsuite/btcd/tree/master/btcec), would prefer to have someone else look over it as well

**Pull request merged**